standard/types.md

```diff
@@ -59,10 +59,7 @@ array_type
 
 non_array_type
     : value_type
-    | class_type
-    | interface_type
-    | delegate_type
-    | 'dynamic'
+    | (class_type | interface_type | delegate_type | 'dynamic') nullable_type_annotation?
     | type_parameter
```

Comment:
I'm realizing `nullable_reference_type` does not include type parameters unless it's via a mechanism I'm not seeing. We'd also want to be able to `T?[]` where `T` is constrained to a class type.